src/main/scala/is/hail/variant/package.scala

```diff
@@ -29,6 +29,16 @@ package object variant {
           override def nextInt(): Int = it.next().unboxedGT
         }
     }
+
+    def dosageIterator: DoubleIterator = ig match {
+      case gs: GenotypeStream => gs.gsDosageIterator
+      case _ =>
+        new DoubleIterator {
```

Comment:
I guess this is also a question for the IntIterator stuff, but why do we have to use the iterator-like hasNext function if we know exactly how many samples / genotypes we have?